pkg/dal/service.go

```diff
@@ -833,12 +835,14 @@ func (svc *service) removeModelFromRegistry(model *Model) {
 // ReplaceModelAttribute adds new or updates an existing attribute for the given model
 //
 // We rely on the user to provide stable and valid attribute definitions.
-func (svc *service) ReplaceModelAttribute(ctx context.Context, model *Model, old, new *Attribute, trans ...TransformationFunction) (err error) {
+func (svc *service) ReplaceModelAttribute(ctx context.Context, model *Model, diff *ModelDiff, hasRecords bool,trans ...TransformationFunction) (err error) {
 	svc.logger.Debug("updating model attribute", zap.Uint64("model", model.ResourceID))
 
 	var (
 		conn   *ConnectionWrap
 		issues = newIssueHelper().addModel(model.ResourceID)
+        _new   = diff.Asserted
```

Comment:
Find a better name that does not require a `_` prefix. 
TBH I don't mind using the full name (diff.Asserted and diff.Original).

Reply:
I'll use the full name everwhere